### PR TITLE
Change bundle dependency from dict to list

### LINKED ISSUE #2021: [USER ISSUE] Multiple aliases pointing to the same bundle spec are not supported
**Describe the bug**
When I set two bundle dependencies with the same bundle spec but different aliases, one of them is get dropped.

**Codalab Information**
- *Your username*: hfang
- *UUID of worksheet where problem occurred*: N/A
- *UUID of problem bundle (please don't delete)*: 0x218c2d796c9847a283817362459eb308
- *Error message received from Codalab (if applicable)*: N/A
 - *OS (if on CLI)*: N/A
 - *Codalab CLI Version (if on CLI)*:  N/A
 - *Browser (if on Web)*; Chrome

**To Reproduce**
Steps to reproduce the behavior :
```
run foo:0xea12e8bbfa9d4e0788dfef6457df7cee/foo foo2:0xea12e8bbfa9d4e0788dfef6457df7cee --- "cat foo2"
```

**Expected behavior**
Both "foo" and "foo2" should be accessible and shown as dependencies.

**Screenshots**
N/A

**Additional context**
This is a sample job to illustrate the problem.
It happens for a complicated worksheet in practice, when sometimes the two dependencies can be different, but sometimes they point to the same bundle spec.

### PR DESCRIPTION
**Pull request by candicegjing**
Fixed #2021 

This PR will support multiple keys map to single value, e.g. `foo:<uuid_1>, foo2:<uuid_1>`. 
A real example could be as follows:
Screen Shot 2020-03-24 at 9 12 38 PM

### CODE REVIEW

File: codalab/worker/bundle_state.py [hunk 1]
```diff
@@ -76,23 +76,27 @@ def __init__(
...
-            DependencyKey(dep["parent_uuid"], dep["parent_path"]): Dependency(
-                parent_name=dep["parent_name"],
-                parent_path=dep["parent_path"],
-                parent_uuid=dep["parent_uuid"],
-                child_path=dep["child_path"],
-                child_uuid=dep["child_uuid"],
-                location=dep.get("location", None),
+        self.dependencies = [
+            (
+                DependencyKey(dep["parent_uuid"], dep["parent_path"]),
```
Comment (percyliang) [inline codalab/worker/bundle_state.py:81]: Don't need the key if it's a list because it's already in `Dependency`

**Review by percyliang — COMMENTED** [review]
`self.dependencies` isn't used for dict lookup anywhere?  I would have assumed that the key was there for a reason...

**Comment by candicegjing** [comment]
> `self.dependencies` isn't used for dict lookup anywhere? I would have assumed that the key was there for a reason...

I searched through our codebase and doubt that it has been used anywhere else other than this file. @bkgoksel Could you please verify in case I overlooked some details?

File: codalab/worker/bundle_state.py (hunk 1 above)
Comment (candicegjing) [reply codalab/worker/bundle_state.py:81]: That's true. I have updated the code.

**Review by percyliang — APPROVED** [review]
LGTM!

**Review by percyliang — APPROVED** [review]

**Comment by candicegjing** [comment]
Close this PR as we decide to go with another solution which is documented at #2115

**Pull request closed without merging**